src/middleware/AuthMiddleware.ts

```diff
@@ -0,0 +1,22 @@
+import { provide } from 'inversify-binding-decorators';
+import { BaseMiddleware } from 'inversify-express-utils';
+import { Request, Response, NextFunction, RequestHandler } from 'express';
+import * as expressAsyncHandler from 'express-async-handler';
+
+@provide(AuthMiddleware)
+export class AuthMiddleware extends BaseMiddleware {
+
+    public constructor() {
+        super();
+    }
+
+    public handler: RequestHandler = expressAsyncHandler((req: Request, res: Response, next: NextFunction): void => {
+
+        if (req.session.isNothing()) {
+            res.redirect(`http://localhost:3001/signin?return_to=${req.url}`);
+        }
+        console.log('Authenticated');
```

Comment:
How do we know at this point the sign in was successful?  Or do we assume it was and then have a check in the controller?

Reply:
Check new version